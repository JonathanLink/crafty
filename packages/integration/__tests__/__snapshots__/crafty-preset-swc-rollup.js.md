# Snapshot report for `__tests__/crafty-preset-swc-rollup.js`

The actual snapshot is saved in `crafty-preset-swc-rollup.js.snap`.

Generated by [AVA](https://avajs.dev).

## Compiles JavaScript with rollup

> Snapshot 1

    {
      status: 0,
      stdall: `␊
      [__:__:__] Starting Crafty __version__...␊
      [__:__:__] Files will be stored in:␊
                  js: __PATH__/packages/integration/fixtures/crafty-preset-swc-rollup/compiles/dist/js␊
      [__:__:__] Starting 'default' ...␊
      [__:__:__] Starting 'js' ...␊
      [__:__:__] Starting 'js_myBundle' ...␊
      [__:__:__] Finished 'js_myBundle' after ____ ms␊
      [__:__:__] Finished 'js' after ____ ms␊
      [__:__:__] Finished 'default' after ____ ms␊
      `,
    }

> Snapshot 2

    `function e(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}class t extends React.Component{render(){/*#__PURE__*/␊
    return React.createElement("div",{onClick:this.handleClick},"Hey ",this.state.hey)}constructor(...t){super(...t),e(this,"handleClick",(()=>{this.setState({hey:!0})}))}}class n{method(){console.log(t)}constructor(){console.log("init class")}}export{n as default};␊
    //# sourceMappingURL=myBundle.min.js.map␊
    `

## Keeps imports unresolved for SWC Runtime

> Snapshot 1

    {
      status: 0,
      stdall: `␊
      [__:__:__] Starting Crafty __version__...␊
      [__:__:__] Files will be stored in:␊
                  js: __PATH__/packages/integration/fixtures/crafty-preset-swc-rollup/compiles-import-runtime/dist/js␊
      [__:__:__] Starting 'default' ...␊
      [__:__:__] Starting 'js' ...␊
      [__:__:__] Starting 'js_myBundle' ...␊
      [__:__:__] Finished 'js_myBundle' after ____ ms␊
      [__:__:__] Finished 'js' after ____ ms␊
      [__:__:__] Finished 'default' after ____ ms␊
      `,
    }

> Snapshot 2

    `import e from"@swc/helpers/src/_class_call_check.mjs";import r from"@swc/helpers/src/_create_class.mjs";import s from"@swc/helpers/src/_assert_this_initialized.mjs";import t from"@swc/helpers/src/_define_property.mjs";import c from"@swc/helpers/src/_inherits.mjs";import i from"@swc/helpers/src/_create_super.mjs";var n=/*#__PURE__*/function(n){c(l,n);var o=i(l);function l(){var r;return e(this,l),r=o.apply(this,arguments),t(s(r),"handleClick",(function(){r.setState({hey:!0})})),r}return r(l,[{key:"render",value:function(){/*#__PURE__*/␊
    return React.createElement("div",{onClick:this.handleClick},"Hey ",this.state.hey)}}]),l}(React.Component);var o=/*#__PURE__*/function(){function s(){e(this,s),console.log("init class")}return r(s,[{key:"method",value:function(){console.log(n)}}]),s}();export{o as default};␊
    //# sourceMappingURL=myBundle.min.js.map␊
    `

## Keeps imports unresolved for SWC Runtime, using commonjs

> Snapshot 1

    {
      status: 0,
      stdall: `␊
      [__:__:__] Starting Crafty __version__...␊
      [__:__:__] Files will be stored in:␊
                  js: __PATH__/packages/integration/fixtures/crafty-preset-swc-rollup/compiles-import-runtime-commonjs/dist/js␊
      [__:__:__] Starting 'default' ...␊
      [__:__:__] Starting 'js' ...␊
      [__:__:__] Starting 'js_myBundle' ...␊
      [__:__:__] Finished 'js_myBundle' after ____ ms␊
      [__:__:__] Finished 'js' after ____ ms␊
      [__:__:__] Finished 'default' after ____ ms␊
      `,
    }

> Snapshot 2

    `"use strict";var e=require("@swc/helpers/src/_class_call_check.mjs");var r=require("@swc/helpers/src/_create_class.mjs");var s=require("@swc/helpers/src/_assert_this_initialized.mjs");var t=require("@swc/helpers/src/_define_property.mjs");var c=require("@swc/helpers/src/_inherits.mjs");var i=require("@swc/helpers/src/_create_super.mjs");var n=/*#__PURE__*/function(n){c(l,n);var a=i(l);function l(){var r;return e(this,l),r=a.apply(this,arguments),t(s(r),"handleClick",(function(){r.setState({hey:!0})})),r}return r(l,[{key:"render",value:function(){/*#__PURE__*/␊
    return React.createElement("div",{onClick:this.handleClick},"Hey ",this.state.hey)}}]),l}(React.Component);var a=/*#__PURE__*/function(){function s(){e(this,s),console.log("init class")}return r(s,[{key:"method",value:function(){console.log(n)}}]),s}();module.exports=a;␊
    //# sourceMappingURL=myBundle.min.js.map␊
    `

## Fails gracefully on broken markup

> Snapshot 1

    {
      status: 1,
      stdall: `␊
      [__:__:__] Starting Crafty __version__...␊
      [__:__:__] Files will be stored in:␊
                  js: __PATH__/packages/integration/fixtures/crafty-preset-swc-rollup/fails/dist/js␊
      [__:__:__] Starting 'default' ...␊
      [__:__:__] Starting 'js' ...␊
      [__:__:__] Starting 'js_myBundle' ...␊
      ␊
      __PATH__/packages/integration/fixtures/crafty-preset-swc-rollup/fails/js/script.js␊
        15:1  error  Parsing error: Unexpected token␊
      ␊
      ✖ 1 problem (1 error, 0 warnings)␊
      ␊
      [js_myBundle] (!) Errors were found␊
      [__:__:__] 'js_myBundle' errored after ____ ms␊
      [__:__:__] Errors were found␊
      [__:__:__] 'js' errored after ____ ms␊
      [__:__:__] 'default' errored after ____ ms␊
      `,
    }

## Lints JavaScript with rollup

> Snapshot 1

    {
      status: 1,
      stdall: `␊
      [__:__:__] Starting Crafty __version__...␊
      [__:__:__] Files will be stored in:␊
                  js: __PATH__/packages/integration/fixtures/crafty-preset-swc-rollup/lints/dist/js␊
      [__:__:__] Starting 'default' ...␊
      [__:__:__] Starting 'js' ...␊
      [__:__:__] Starting 'js_myBundle' ...␊
      ␊
      __PATH__/packages/integration/fixtures/crafty-preset-swc-rollup/lints/js/Component.js␊
         4:1   error  Delete \`⏎\`                                 @swissquote/swissquote/prettier/prettier␊
         6:20  error  Replace \`hey:·true⏎···\` with \`·hey:·true\`  @swissquote/swissquote/prettier/prettier␊
        10:11  error  Delete \`⏎·\`                                @swissquote/swissquote/prettier/prettier␊
        12:11  error  Delete \`·\`                                 @swissquote/swissquote/prettier/prettier␊
      ␊
      __PATH__/packages/integration/fixtures/crafty-preset-swc-rollup/lints/js/script.js␊
         3:23  error    Delete \`·\`                                       @swissquote/swissquote/prettier/prettier␊
         5:10  error    'test' is defined but never used                 no-unused-vars␊
         5:14  error    Replace \`·(one,\` with \`(one,·\`                   @swissquote/swissquote/prettier/prettier␊
         6:1   error    Delete \`·\`                                       @swissquote/swissquote/prettier/prettier␊
        11:5   warning  Unexpected console statement                     no-console␊
        14:9   error    Delete \`·\`                                       @swissquote/swissquote/prettier/prettier␊
        15:5   warning  Unexpected console statement                     no-console␊
        15:17  error    Replace \`··"some·method"·\` with \`"some·method"\`  @swissquote/swissquote/prettier/prettier␊
        18:11  error    Replace \`·a,·b)⏎·\` with \`a,·b)\`                  @swissquote/swissquote/prettier/prettier␊
        20:14  error    Expected '===' and instead saw '=='              eqeqeq␊
      ␊
      ✖ 14 problems (12 errors, 2 warnings)␊
        10 errors and 0 warnings potentially fixable with the \`--fix\` option.␊
      ␊
      [js_myBundle] (!) Errors were found␊
      [__:__:__] 'js_myBundle' errored after ____ ms␊
      [__:__:__] Errors were found␊
      [__:__:__] 'js' errored after ____ ms␊
      [__:__:__] 'default' errored after ____ ms␊
      `,
    }
