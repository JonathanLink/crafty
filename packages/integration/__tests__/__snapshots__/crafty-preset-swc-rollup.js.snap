// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compiles JavaScript with rollup 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-swc-rollup/compiles/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Compiles JavaScript with rollup 2`] = `
"function e(e){if(void 0===e)throw new ReferenceError(\\"this hasn't been initialised - super() hasn't been called\\");return e}function t(e,t){if(!(e instanceof t))throw new TypeError(\\"Cannot call a class as a function\\")}function n(e,t){for(var n=0;n<t.length;n++){var r=t[n];r.enumerable=r.enumerable||!1,r.configurable=!0,\\"value\\"in r&&(r.writable=!0),Object.defineProperty(e,r.key,r)}}function r(e,t,r){return t&&n(e.prototype,t),r&&n(e,r),e}function o(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function i(e){return i=Object.setPrototypeOf?Object.getPrototypeOf:function(e){return e.__proto__||Object.getPrototypeOf(e)},i(e)}function u(e,t){if(\\"function\\"!==typeof t&&null!==t)throw new TypeError(\\"Super expression must either be null or a function\\");e.prototype=Object.create(t&&t.prototype,{constructor:{value:e,writable:!0,configurable:!0}}),t&&a(e,t)}function c(t,n){if(n&&(\\"object\\"===f(n)||\\"function\\"===typeof n))return n;return e(t)}function a(e,t){return a=Object.setPrototypeOf||function(e,t){return e.__proto__=t,e},a(e,t)}var f=function(e){return e&&\\"undefined\\"!==typeof Symbol&&e.constructor===Symbol?\\"symbol\\":typeof e};var l=function(n){function a(){var n;return t(this,a),n=c(this,i(a).apply(this,arguments)),o(e(n),\\"handleClick\\",(function(){n.setState({hey:!0})})),n}return u(a,n),r(a,[{key:\\"render\\",value:function(){return React.createElement(\\"div\\",{onClick:this.handleClick},\\"Hey \\",this.state.hey)}}]),a}(React.Component);function s(e,t){if(!(e instanceof t))throw new TypeError(\\"Cannot call a class as a function\\")}function p(e,t){for(var n=0;n<t.length;n++){var r=t[n];r.enumerable=r.enumerable||!1,r.configurable=!0,\\"value\\"in r&&(r.writable=!0),Object.defineProperty(e,r.key,r)}}function y(e,t,n){return t&&p(e.prototype,t),n&&p(e,n),e}var b=function(){function e(){s(this,e),console.log(\\"init class\\")}return y(e,[{key:\\"method\\",value:function(){console.log(l)}}]),e}();export default b;
//# sourceMappingURL=myBundle.min.js.map
"
`;

exports[`Fails gracefully on broken markup 1`] = `
Object {
  "status": 1,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-swc-rollup/fails/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
error: Expected }, got <eof>
  --> __PATH__/packages/integration/fixtures/crafty-preset-swc-rollup/fails/js/script.js:14:1
   |
14 | }
   | ^

__PATH__/packages/integration/fixtures/crafty-preset-swc-rollup/fails/js/script.js
  15:1  error  Parsing error: Unexpected token

  13 |
  14 | }
> 15 |
     | ^

✖ 1 problem (1 error, 0 warnings)

[js_myBundle] (!) Errors were found
[__:__:__] 'js_myBundle' errored after ____ ms
[__:__:__] Error: Errors were found
    ...stacktrace...
[__:__:__] 'js' errored after ____ ms
[__:__:__] 'default' errored after ____ ms
",
}
`;

exports[`Keeps imports unresolved for SWC Runtime 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-swc-rollup/compiles-import-runtime/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Keeps imports unresolved for SWC Runtime 2`] = `
"import*as e from\\"@swc/helpers\\";var t=function(t){function n(){var t;return e.classCallCheck(this,n),t=e.possibleConstructorReturn(this,e.getPrototypeOf(n).apply(this,arguments)),e.defineProperty(e.assertThisInitialized(t),\\"handleClick\\",(function(){t.setState({hey:!0})})),t}return e.inherits(n,t),e.createClass(n,[{key:\\"render\\",value:function(){return React.createElement(\\"div\\",{onClick:this.handleClick},\\"Hey \\",this.state.hey)}}]),n}(React.Component);var n=function(){function n(){e.classCallCheck(this,n),console.log(\\"init class\\")}return e.createClass(n,[{key:\\"method\\",value:function(){console.log(t)}}]),n}();export default n;
//# sourceMappingURL=myBundle.min.js.map
"
`;

exports[`Keeps imports unresolved for SWC Runtime, using commonjs 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-swc-rollup/compiles-import-runtime-commonjs/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
Entry module \\"js/script.js\\" is implicitly using \\"default\\" export mode, which means for CommonJS output that its default export is assigned to \\"module.exports\\". For many tools, such CommonJS output will not be interchangeable with the original ES module. If this is intended, explicitly set \\"output.exports\\" to either \\"auto\\" or \\"default\\", otherwise you might want to consider changing the signature of \\"js/script.js\\" to use named exports only.
[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Keeps imports unresolved for SWC Runtime, using commonjs 2`] = `
"\\"use strict\\";var e=require(\\"@swc/helpers\\");function t(e){if(e&&e.__esModule)return e;var t=Object.create(null);return e&&Object.keys(e).forEach((function(r){if(\\"default\\"!==r){var n=Object.getOwnPropertyDescriptor(e,r);Object.defineProperty(t,r,n.get?n:{enumerable:!0,get:function(){return e[r]}})}})),t.default=e,Object.freeze(t)}var r=/*#__PURE__*/t(e);var n=function(e){function t(){var e;return r.classCallCheck(this,t),e=r.possibleConstructorReturn(this,r.getPrototypeOf(t).apply(this,arguments)),r.defineProperty(r.assertThisInitialized(e),\\"handleClick\\",(function(){e.setState({hey:!0})})),e}return r.inherits(t,e),r.createClass(t,[{key:\\"render\\",value:function(){return React.createElement(\\"div\\",{onClick:this.handleClick},\\"Hey \\",this.state.hey)}}]),t}(React.Component);var c=function(){function e(){r.classCallCheck(this,e),console.log(\\"init class\\")}return r.createClass(e,[{key:\\"method\\",value:function(){console.log(n)}}]),e}();module.exports=c;
//# sourceMappingURL=myBundle.min.js.map
"
`;

exports[`Lints JavaScript with rollup 1`] = `
Object {
  "status": 1,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-swc-rollup/lints/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...

__PATH__/packages/integration/fixtures/crafty-preset-swc-rollup/lints/js/Component.js
   4:1   error  Delete \`⏎\`                                 @swissquote/swissquote/prettier/prettier
   6:20  error  Replace \`hey:·true⏎···\` with \`·hey:·true\`  @swissquote/swissquote/prettier/prettier
  10:11  error  Delete \`⏎·\`                                @swissquote/swissquote/prettier/prettier
  12:11  error  Delete \`·\`                                 @swissquote/swissquote/prettier/prettier

__PATH__/packages/integration/fixtures/crafty-preset-swc-rollup/lints/js/script.js
   3:23  error    Delete \`·\`                                       @swissquote/swissquote/prettier/prettier
   5:10  error    'test' is defined but never used                 no-unused-vars
   5:14  error    Replace \`·(one,\` with \`(one,·\`                   @swissquote/swissquote/prettier/prettier
   6:1   error    Delete \`·\`                                       @swissquote/swissquote/prettier/prettier
  11:5   warning  Unexpected console statement                     no-console
  14:9   error    Delete \`·\`                                       @swissquote/swissquote/prettier/prettier
  15:5   warning  Unexpected console statement                     no-console
  15:17  error    Replace \`··\\"some·method\\"·\` with \`\\"some·method\\"\`  @swissquote/swissquote/prettier/prettier
  18:11  error    Replace \`·a,·b)⏎·\` with \`a,·b)\`                  @swissquote/swissquote/prettier/prettier
  20:14  error    Expected '===' and instead saw '=='              eqeqeq

✖ 14 problems (12 errors, 2 warnings)
  10 errors and 0 warnings potentially fixable with the \`--fix\` option.

[js_myBundle] (!) Errors were found
[__:__:__] 'js_myBundle' errored after ____ ms
[__:__:__] Error: Errors were found
    ...stacktrace...
[__:__:__] 'js' errored after ____ ms
[__:__:__] 'default' errored after ____ ms
",
}
`;
